File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n <= 1:
            return n
        if n == 2:
            return 1
        t = [0] * (n+1)
        t[1], t[2] = 1, 1
        for i in range(3, n+1):
            t[i] = t[i-1] + t[i-2] + t[i-3]
        return t[n]

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        vector<int> tribo(n+1, 0);
        
        if (n == 0) return 0;
        if (n == 1 || n == 2) return 1;

        tribo[1] = 1;
        tribo[2] = 1;

        for (int i = 3; i < n + 1; i++) {
            tribo[i] = tribo[i - 1] + tribo[i - 2] + tribo[i - 3];
        }
        return tribo[n];
    }
};